code/Controller/AssetAdminOpen.php

```diff
@@ -411,9 +411,19 @@ private function filterFiles(DataList $list, array $filter): DataList
             $list = $list->filter("Created:LessThanOrEqual", $toDate->dataValue().' 23:59:59');
             $search = true;
         }
-        // Categories (mapped to extensions through the enum type automatically)
+        // Categories (mapped to extensions)
         if (!empty($filter['appCategory'])) {
-            $list = $list->filter('Name:EndsWith', $filter['appCategory']);
+            $categoryValues = File::config()->get('app_categories');
+            // Sanitise category names (some contain slashes) and make all upper case
+            foreach ($categoryValues as $key => $extensions) {
+                unset($categoryValues[$key]);
+                $newKey = strtoupper(preg_replace('/[^[[:alnum:]]]*/', '', $key));
```

Comment:
```suggestion
                $newKey = strtoupper(preg_replace('/[^a-zA-Z0-9]/', '', $key));
```
We don't use `[:alnum:]` anywhere else in the codebase. I had to lookup what exact characters it represented. I think it's better to just explicitly list the chars.  We can also get rid of 2 sets of brackets and the asterisk this way